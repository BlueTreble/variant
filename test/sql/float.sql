\set ECHO 0
\set QUIET 1
/*
 * NOTE! Unlike the other cases of duplication, we need to handle numeric,
 * float and real separately because of odd things such as
 *
 * 0.0::float < 0.0::real = true
 */


\set baseline_type float
-- Space separated string
\set test_types 'float'
-- Used in array[ :base_values ]
\set base_values '-1.1, 0.1, 1.1, NULL' 
\set compare_value 0.1


\i test/helpers/type_setup.sql

SELECT plan(sum(test_count)::int) FROM plan;

\i test/helpers/type.sql


-- vi: expandtab sw=2 ts=2
