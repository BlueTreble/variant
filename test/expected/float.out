\set ECHO 0


ok 1..0


1..90
ok 1 - No NULL variants
ok 2 - Verify result of cast '-1.1'::float
ok 3 - Verify result of cast '0.1'::float
ok 4 - Verify result of cast '1.1'::float
ok 5 - Verify result of cast NULL::float
ok 6 - SET base_variant_b
ok 7 - Verify text_in(text_out()) works for ("double precision",-1.1)
ok 8 - Verify text_in(text_out()) works for ("double precision",0.1)
ok 9 - Verify text_in(text_out()) works for ("double precision",1.1)
ok 10 - Verify text_in(text_out()) works for ("double precision",)
ok 11
ok 12
ok 13
ok 14
ok 15 - No NULL variants
ok 16 - Verify result of cast '0.1'::float
ok 17 - SET compare_variant_b
ok 18 - Verify text_in(text_out()) works for ("double precision",0.1)
ok 19 - check casted vs uncasted for '0.1'::float <   '-1.1'::float
ok 20 - check casted vs uncasted for '0.1'::float <   '0.1'::float
ok 21 - check casted vs uncasted for '0.1'::float <   '1.1'::float
ok 22 - check casted vs uncasted for '0.1'::float <   NULL::float
ok 23 - check casted vs uncasted for '0.1'::float <=  '-1.1'::float
ok 24 - check casted vs uncasted for '0.1'::float <=  '0.1'::float
ok 25 - check casted vs uncasted for '0.1'::float <=  '1.1'::float
ok 26 - check casted vs uncasted for '0.1'::float <=  NULL::float
ok 27 - check casted vs uncasted for '0.1'::float  =  '-1.1'::float
ok 28 - check casted vs uncasted for '0.1'::float  =  '0.1'::float
ok 29 - check casted vs uncasted for '0.1'::float  =  '1.1'::float
ok 30 - check casted vs uncasted for '0.1'::float  =  NULL::float
ok 31 - check casted vs uncasted for '0.1'::float  >= '-1.1'::float
ok 32 - check casted vs uncasted for '0.1'::float  >= '0.1'::float
ok 33 - check casted vs uncasted for '0.1'::float  >= '1.1'::float
ok 34 - check casted vs uncasted for '0.1'::float  >= NULL::float
ok 35 - check casted vs uncasted for '0.1'::float   > '-1.1'::float
ok 36 - check casted vs uncasted for '0.1'::float   > '0.1'::float
ok 37 - check casted vs uncasted for '0.1'::float   > '1.1'::float
ok 38 - check casted vs uncasted for '0.1'::float   > NULL::float
ok 39 - check casted vs uncasted for '0.1'::float !=  '-1.1'::float
ok 40 - check casted vs uncasted for '0.1'::float !=  '0.1'::float
ok 41 - check casted vs uncasted for '0.1'::float !=  '1.1'::float
ok 42 - check casted vs uncasted for '0.1'::float !=  NULL::float
ok 43 - check variant vs uncasted for '0.1'::float <   '-1.1'::float
ok 44 - check variant vs uncasted for '0.1'::float <   '0.1'::float
ok 45 - check variant vs uncasted for '0.1'::float <   '1.1'::float
ok 46 - check variant vs uncasted for '0.1'::float <   NULL::float
ok 47 - check variant vs uncasted for '0.1'::float <=  '-1.1'::float
ok 48 - check variant vs uncasted for '0.1'::float <=  '0.1'::float
ok 49 - check variant vs uncasted for '0.1'::float <=  '1.1'::float
ok 50 - check variant vs uncasted for '0.1'::float <=  NULL::float
ok 51 - check variant vs uncasted for '0.1'::float  =  '-1.1'::float
ok 52 - check variant vs uncasted for '0.1'::float  =  '0.1'::float
ok 53 - check variant vs uncasted for '0.1'::float  =  '1.1'::float
ok 54 - check variant vs uncasted for '0.1'::float  =  NULL::float
ok 55 - check variant vs uncasted for '0.1'::float  >= '-1.1'::float
ok 56 - check variant vs uncasted for '0.1'::float  >= '0.1'::float
ok 57 - check variant vs uncasted for '0.1'::float  >= '1.1'::float
ok 58 - check variant vs uncasted for '0.1'::float  >= NULL::float
ok 59 - check variant vs uncasted for '0.1'::float   > '-1.1'::float
ok 60 - check variant vs uncasted for '0.1'::float   > '0.1'::float
ok 61 - check variant vs uncasted for '0.1'::float   > '1.1'::float
ok 62 - check variant vs uncasted for '0.1'::float   > NULL::float
ok 63 - check variant vs uncasted for '0.1'::float !=  '-1.1'::float
ok 64 - check variant vs uncasted for '0.1'::float !=  '0.1'::float
ok 65 - check variant vs uncasted for '0.1'::float !=  '1.1'::float
ok 66 - check variant vs uncasted for '0.1'::float !=  NULL::float
ok 67 - check variant vs casted for '''0.1''::float' <   '''-1.1''::float'
ok 68 - check variant vs casted for '''0.1''::float' <   '''0.1''::float'
ok 69 - check variant vs casted for '''0.1''::float' <   '''1.1''::float'
ok 70 - check variant vs casted for '''0.1''::float' <   'NULL::float'
ok 71 - check variant vs casted for '''0.1''::float' <=  '''-1.1''::float'
ok 72 - check variant vs casted for '''0.1''::float' <=  '''0.1''::float'
ok 73 - check variant vs casted for '''0.1''::float' <=  '''1.1''::float'
ok 74 - check variant vs casted for '''0.1''::float' <=  'NULL::float'
ok 75 - check variant vs casted for '''0.1''::float'  =  '''-1.1''::float'
ok 76 - check variant vs casted for '''0.1''::float'  =  '''0.1''::float'
ok 77 - check variant vs casted for '''0.1''::float'  =  '''1.1''::float'
ok 78 - check variant vs casted for '''0.1''::float'  =  'NULL::float'
ok 79 - check variant vs casted for '''0.1''::float'  >= '''-1.1''::float'
ok 80 - check variant vs casted for '''0.1''::float'  >= '''0.1''::float'
ok 81 - check variant vs casted for '''0.1''::float'  >= '''1.1''::float'
ok 82 - check variant vs casted for '''0.1''::float'  >= 'NULL::float'
ok 83 - check variant vs casted for '''0.1''::float'   > '''-1.1''::float'
ok 84 - check variant vs casted for '''0.1''::float'   > '''0.1''::float'
ok 85 - check variant vs casted for '''0.1''::float'   > '''1.1''::float'
ok 86 - check variant vs casted for '''0.1''::float'   > 'NULL::float'
ok 87 - check variant vs casted for '''0.1''::float' !=  '''-1.1''::float'
ok 88 - check variant vs casted for '''0.1''::float' !=  '''0.1''::float'
ok 89 - check variant vs casted for '''0.1''::float' !=  '''1.1''::float'
ok 90 - check variant vs casted for '''0.1''::float' !=  'NULL::float'
